src/ToIdn.php

```diff
@@ -0,0 +1,66 @@
+<?php
+namespace Algo26\IdnaConvert;
+
+use Algo26\IdnaConvert\Exception\InvalidCharacterException;
+use Algo26\IdnaConvert\Exception\InvalidIdnVersionException;
+use Algo26\IdnaConvert\Punycode\ToPunycode;
+use Algo26\IdnaConvert\TranscodeUnicode\TranscodeUnicode;
+
+class ToIdn extends AbstractIdnaConvert implements IdnaConvertInterface
+{
+    /** @var TranscodeUnicode */
+    private $unicodeTransCoder;
+
+    /** @var ToPunycode */
+    private $punycodeEncoder;
+
+    /**
+     * @throws InvalidIdnVersionException
+     */
+    public function __construct($idnVersion = null)
+    {
+        $this->unicodeTransCoder = new TranscodeUnicode();
+        $this->punycodeEncoder = new ToPunycode($idnVersion);
+    }
+
+    /**
+     * @param string $host
+     *
+     * @return string
+     * @throws InvalidCharacterException
+     * @throws Exception\AlreadyPunycodeException
```

Comment:
Nowhere else is the Exception subnamespace given for the @throws directive

Reply:
I think as long as it's resolved correctly (`Exception` namespace imported), it  does not matter what style you use (importing classes or importing namespaces).